lib/features/thread/data/network/thread_api.dart

```diff
@@ -83,6 +87,86 @@ class ThreadAPI {
     return EmailsResponse(emailList: resultList?.list, state: resultList?.state);
   }
 
+  Future<SearchEmailsResponse> searchEmails(
+    Session session,
+    AccountId accountId,
+    {
+      UnsignedInt? limit,
+      int? position,
+      Set<Comparator>? sort,
+      Filter? filter,
+      Properties? properties
+    }
+  ) async {
+    final processingInvocation = ProcessingInvocation();
+
+    final jmapRequestBuilder = JmapRequestBuilder(httpClient, processingInvocation);
+
+    // Email/query
+    final queryEmailMethod = QueryEmailMethod(accountId);
+
+    if (limit != null) queryEmailMethod.addLimit(limit);
+
+    if (position != null) queryEmailMethod.addPosition(position);
+
+    if (sort != null) queryEmailMethod.addSorts(sort);
+
+    if (filter != null) queryEmailMethod.addFilters(filter);
+
+    final queryEmailInvocation = jmapRequestBuilder.invocation(queryEmailMethod);
+
+    // SearchSnippet/get
+    final getSearchSnippetMethod = SearchSnippetGetMethod(accountId);
+    if (filter != null) getSearchSnippetMethod.addFilters(filter);
+    getSearchSnippetMethod.addReferenceEmailIds(
+      processingInvocation.createResultReference(
+        queryEmailInvocation.methodCallId,
+        ReferencePath.idsPath));
+    final getSearchSnippetInvocation = jmapRequestBuilder.invocation(
+      getSearchSnippetMethod);
+
+    // Email/get
+    final getEmailMethod = GetEmailMethod(accountId);
+
+    if (properties != null) getEmailMethod.addProperties(properties);
+
+    getEmailMethod.addReferenceIds(processingInvocation.createResultReference(
+      queryEmailInvocation.methodCallId,
+      ReferencePath.idsPath));
+
+    final getEmailInvocation = jmapRequestBuilder.invocation(getEmailMethod);
+
+    final capabilities = getEmailMethod.requiredCapabilities
+      .toCapabilitiesSupportTeamMailboxes(session, accountId);
+
+    final result = await (jmapRequestBuilder
+        ..usings(capabilities))
+      .build()
+      .execute();
+
+    final emailResultList = result.parse<GetEmailResponse>(
+        getEmailInvocation.methodCallId, GetEmailResponse.deserialize);
+
+    if (sort != null && emailResultList != null) {
+      for (var comparator in sort) {
+        emailResultList.sortEmails(comparator);
+      }
+    }
+
+    SearchSnippetGetResponse? searchSnippetGetResponse;
+    try {
+      searchSnippetGetResponse = result.parse<SearchSnippetGetResponse>(
+          getSearchSnippetInvocation.methodCallId,
+          SearchSnippetGetResponse.fromJson);
+    } catch (e) {
+      logError('ThreadAPI::searchEmails:searchSnippetGetResponse: Exception = $e');
+    }
```

Comment:
Should separate function to handle it. also write  write unit for it

Reply:
Done